### PR TITLE
Add Seed Based `Wallet` Derivation to `WalletFactory`

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Add a method to generate a `Wallet` from a seed on `WalletFactory`. 

### Context of Change

This PR continues to deprecate static factory methods on `Wallet` and replace them with a properly thought out class, `WalletFactory`. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Additional functionality is placed on WalletFactory. 

## Test Plan

CI; additional tests are provided for new functionality.  

## Future Tasks
- Continue to migrate functionality to `WalletFactory`. 
- Remove deprecated methods in N+2 releases.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -67,43 +82,50 @@ Update to the latest version of protocol buffers from rippled, introduced in [#3
...
 
 ### Fixed
+
 - Build and Webpack output now include the generated protobuf typescript type definintions.
 
 ## [3.0.0] - 2019-02-24
 
-This release adds support for (protocol buffers in rippled)[https://github.com/ripple/rippled/tree/develop/src/ripple/proto/rpc/v1]. The `Signer`'s `signTransaction` method previously returned a hex string representing the signature of the transaction. After this change, it returns the entire signed and encoded transaction as a byte array. This encapsulates the serialization and signing logic inside of this library, rather than forcing clients to understand it. This is a breaking change (See 'Breaking Changes') below. 
+This release adds support for (protocol buffers in rippled)[https://github.com/ripple/rippled/tree/develop/src/ripple/proto/rpc/v1]. The `Signer`'s `signTransaction` method previously returned a hex string representing the signature of the transaction. After this change, it returns the entire signed and encoded transaction as a byte array. This encapsulates the serialization and signing logic inside of this library, rather than forcing clients to understand it. This is a breaking change (See 'Breaking Changes') below.
```
Comment (LoisRP) [inline CHANGELOG.md:102]: below should probably be inside the parentheses.

File: CHANGELOG.md [hunk 2]
```diff
@@ -67,43 +82,50 @@ Update to the latest version of protocol buffers from rippled, introduced in [#3
...
 
-This release also changes the protocol buffer compiler used to generate protocol buffers in order to make this library function inside of a browser environment. As part of this change, various optimizations to TypeScript and webpack compilation are also produced (see 'Breaking Changes') below. 
+This release also changes the protocol buffer compiler used to generate protocol buffers in order to make this library function inside of a browser environment. As part of this change, various optimizations to TypeScript and webpack compilation are also produced (see 'Breaking Changes') below.
```
Comment (LoisRP) [inline CHANGELOG.md:104]: placement of below

File: CHANGELOG.md [hunk 3]
```diff
@@ -67,43 +82,50 @@ Update to the latest version of protocol buffers from rippled, introduced in [#3
...
 ### Internal Changes
+
 - Updated the ESLint and TSConfig to properly lint the project to our preferred styles.
 
 ## [2.0.1] - 2019-02-15
 
 ### Fixed
 
-This patch release locks the versions of `grpc-tools` and `grpc_tools_node_protoc_ts` dependencies. This allows client projects to be able to deterministicly generate compatible protocol buffers with this project's TypeScript. 
+This patch release locks the versions of `grpc-tools` and `grpc_tools_node_protoc_ts` dependencies. This allows client projects to be able to deterministicly generate compatible protocol buffers with this project's TypeScript.
```
Comment (LoisRP) [inline CHANGELOG.md:128]: deterministically

File: CHANGELOG.md [hunk 4]
```diff
@@ -115,7 +137,7 @@ The release migrates the protocol buffers in (xpring-common-protocol-buffers)[ht
 
 This release adds support for (protocol buffers in rippled)[https://github.com/ripple/rippled/tree/develop/src/ripple/proto/rpc/v1]. These protocol buffers are the recommended alternative.
 
-The protocol buffers from `rippled` are not compatible with the protocol buffers from `xpring-common-protocol-buffers`. That makes this a *breaking change*. Clients will need to migrate to new methods (see `breaking changes` below).
+The protocol buffers from `rippled` are not compatible with the protocol buffers from `xpring-common-protocol-buffers`. That makes this a _breaking change_. Clients will need to migrate to new methods (see `breaking changes` below).
```
Comment (LoisRP) [inline CHANGELOG.md:140]: I think this is the correct format that should be copied in the other files.

**Review by LoisRP — APPROVED** [review]
Approved, but fix typos and other issues.

**Review by amiecorso — APPROVED** [review]

**Pull request merged**